backend/dataall/core/permissions/services/tenant_policy_service.py

```diff
@@ -1,20 +1,13 @@
-import logging
+from dataall.core.permissions.db.tenant.tenant_policy_repositories import TenantPolicy
+from dataall.core.permissions.constants import permissions
+from dataall.core.permissions.db.permission.permission_repositories import Permission
+from dataall.core.permissions.api.enums import PermissionType
+from dataall.base.db import exceptions
+from dataall.core.permissions.db.tenant.tenant_models import TenantPolicy as TenantPolicyModel, TenantPolicyPermission
+from dataall.core.permissions.db.tenant.tenant_repositories import Tenant as TenantService
 
-from sqlalchemy.sql import and_
 
-from dataall.core.permissions.db.permission_models import PermissionType
-from dataall.base.db import exceptions, paginate
-from dataall.core.permissions import permissions
-from dataall.core.permissions.db import permission_models as models
-from dataall.core.permissions.db.permission_repositories import Permission
-from dataall.core.permissions.db.tenant_repositories import Tenant as TenantService
-
-logger = logging.getLogger(__name__)
-
-TENANT_NAME = 'dataall'
-
-
-class TenantPolicy:
+class TenantPolicyValidationService:
```

Comment:
Although I would make a small change to differentiate the methods that validate input parameters (e.g. `validate_find_tenant_policy`) and those that validate the permissions of the user (e.g. `validate_admin_access`). 
We have 2 options. First we could clearly define the function names so that it is clear the type of validation. For the above case, `validate_find_tenant_policy` ---> `validate_find_tenant_policy_params`. Or option 2 (which I think I prefer) is to split the `TenantPolicyValidationService` into 2 separate classes: `RequestValidator` with the validation of the input parameters and `TenantPolicyValidationService` with the service that validates user permissions. What are your thoughts?

Reply:
I like splitting in two. I'm totally totally the person, who has a bag with bags and a box with boxes))

Reply:
matryoshka matrioska